internal/util/helpers_test.go

```diff
@@ -4,18 +4,767 @@
+
+			// Create the minder directory inside the temp directory
+			minderDir := filepath.Join(tempDir, "minder")
+			err := os.MkdirAll(minderDir, 0750)
+			if err != nil {
+				t.Fatalf("failed to create minder directory: %v", err)
+			}
+
+			filePath := filepath.Join(minderDir, "credentials.json")
+
+			if tt.fileContent != "" {
+				// Create a temporary file with the specified content
+				err := os.WriteFile(filePath, []byte(tt.fileContent), 0600)
+				if err != nil {
+					t.Fatalf("failed to write test file: %v", err)
+				}
+				// Print the file path and content for debugging
+				t.Logf("Test %s: written file path %s with content: %s", tt.name, filePath, tt.fileContent)
+			} else {
+				// Print the file path for debugging
+				t.Logf("Test %s: file path %s not created as file content is empty", tt.name, filePath)
+			}
+
+			// Temporarily override the environment variable for the test
+			originalEnv := os.Getenv("XDG_CONFIG_HOME")
+			err = os.Setenv("XDG_CONFIG_HOME", tempDir)
+			if err != nil {
+				t.Errorf("error setting XDG_CONFIG_HOME: %v", err)
+			}
+
+			result, err := util.LoadCredentials()
+			if tt.expectedError != "" {
+				if err == nil || !strings.HasPrefix(err.Error(), tt.expectedError) {
+					t.Errorf("expected error matching %v, got %v", tt.expectedError, err)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+				if result.AccessToken != tt.expectedResult.AccessToken || result.RefreshToken != tt.expectedResult.RefreshToken || !result.AccessTokenExpiresAt.Equal(tt.expectedResult.AccessTokenExpiresAt) {
+					t.Errorf("expected result %v, got %v", tt.expectedResult, result)
+				}
+			}
+			defer os.Setenv("XDG_CONFIG_HOME", originalEnv)
+			defer os.RemoveAll(tempDir)
+		})
+
+	}
+}
+
+// TestRevokeToken tests the RevokeToken function
+func TestRevokeToken(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name         string
+		token        string
+		issuerUrl    string
+		clientId     string
+		tokenHint    string
+		expectedPath string
+		expectError  bool
+	}{
+		{
+			name:         "Valid token revocation",
+			token:        "test-token",
+			issuerUrl:    "http://localhost:8081",
+			clientId:     "minder-cli",
+			tokenHint:    "refresh_token",
+			expectedPath: "/realms/stacklok/protocol/openid-connect/revoke",
+			expectError:  false,
+		},
+		{
+			name:         "Invalid issuer URL",
+			token:        "test-token",
+			issuerUrl:    "://invalid-url",
+			clientId:     "minder-cli",
+			tokenHint:    "refresh_token",
+			expectedPath: "",
+			expectError:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			var server *httptest.Server
+			if tt.name != "Invalid issuer URL" {
+				server = httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+					if r.URL.Path != tt.expectedPath {
+						t.Errorf("expected path %s, got %s", tt.expectedPath, r.URL.Path)
+					}
```

Comment:
We use [require](https://pkg.go.dev/github.com/stretchr/testify/require) in other parts of the codebase to write this as:

```golang
require.Equal(t, tt.expectedPath, r.URL.Path)
```

You're not required to do so, but it (and `NoError`) can be helpful sometimes.

Reply:
That is good to learn. I changed accordingly.